tests/ui_tools/test_boxes.py

```diff
@@ -68,6 +68,51 @@ def test_not_calling_send_private_message_without_recipients(self, key,
 
         assert not write_box.model.send_private_message.called
 
+    @pytest.mark.parametrize('key', keys_for_command('SAVE_AS_DRAFT'))
+    @pytest.mark.parametrize('stream_message', [True, False])
+    @pytest.mark.parametrize('draft_message_in_current_session', [
+        True, None,
+    ], ids=[
+        'different_saved_draft_in_current_session',
+        'no_saved_draft_in_current_session',
+    ])
+    def test_keypress_SAVE_DRAFT_MESSAGE(self, key, mocker, stream_message,
+                                         draft_message_in_current_session,
+                                         write_box, widget_size):
+        mocker.patch(BOXES + '.WriteBox.update_recipient_emails')
+        (write_box.model.session_draft_message.
+            return_value) = draft_message_in_current_session
+        write_box.stream_write_box = mocker.Mock(edit_text="Current Stream")
+        write_box.title_write_box = mocker.Mock(edit_text="Topic")
+        write_box.msg_write_box = mocker.Mock(edit_text="Random message")
+        write_box.recipient_emails = ['emailgateway@zulip.com']
+
+        if stream_message:
+            write_box.stream_id = 1
+            draft = {
+                'type': 'stream',
+                'to': 'Current Stream',
+                'content': 'Random message',
+                'subject': 'Topic'
+            }
+        else:
+            write_box.to_write_box = mocker.Mock()
+            draft = {
+                'type': 'private',
+                'to': ['emailgateway@zulip.com'],
+                'content': 'Random message'
+            }
```

Comment:
Maybe a `parametrize` could be replaced with this conditional check?

Reply:
Thanks for the feedback @Ezio-Sarthak . I have parametrized the `draft_message` now.